src/components/Graphs/Graph.tsx

```diff
@@ -92,37 +93,56 @@ const Graph: React.FC = () => {
           <Background />
-          <Panel position="top-left">
-            <FlexBox alignItems={FlexBoxAlignItems.Center} role="radiogroup">
-              <fieldset className={styles.fieldsetReset}>
-                <div className={styles.graphHeader}>
-                  <span className={styles.colorizedTitle}>{t('Graphs.colorizedTitle')}</span>
-                  <RadioButton
-                    name="colorBy"
-                    text={t('Graphs.colorsProviderConfig')}
-                    checked={colorBy === 'provider'}
-                    onChange={() => setColorBy('provider')}
-                  />
-                  <RadioButton
-                    name="colorBy"
-                    text={t('Graphs.colorsProvider')}
-                    checked={colorBy === 'source'}
-                    onChange={() => setColorBy('source')}
-                  />
-                  <RadioButton
-                    name="colorBy"
-                    text={t('Graphs.colorsFlux')}
-                    checked={colorBy === 'flux'}
-                    onChange={() => setColorBy('flux')}
-                  />
-                </div>
-              </fieldset>
-            </FlexBox>
-          </Panel>
-          <Panel position="top-right">
-            <Legend legendItems={legendItems} />
-          </Panel>
         </ReactFlow>
+
+        <div className={styles.topLegendContainer}>
+          <Legend legendItems={legendItems} />
+          <Popover
+            opener="filter-button"
+            open={filterPopoverOpen}
+            placement="Bottom"
+            onClose={() => setFilterPopoverOpen(false)}
+          >
+            <div className={styles.popoverButtonContainer}>
+              <Button
+                design={colorBy === 'source' ? 'Emphasized' : 'Default'}
+                onClick={() => {
+                  setColorBy('source');
+                  setFilterPopoverOpen(false);
+                }}
+              >
+                {t('Graphs.colorsProvider')}
+              </Button>
+              <Button
+                design={colorBy === 'provider' ? 'Emphasized' : 'Default'}
+                onClick={() => {
+                  setColorBy('provider');
+                  setFilterPopoverOpen(false);
+                }}
+              >
+                {t('Graphs.colorsProviderConfig')}
+              </Button>
+              <Button
+                design={colorBy === 'flux' ? 'Emphasized' : 'Default'}
+                onClick={() => {
+                  setColorBy('flux');
+                  setFilterPopoverOpen(false);
+                }}
+              >
+                {t('Graphs.colorsFlux')}
+              </Button>
+            </div>
+          </Popover>
+          <div className={styles.filterIcon}>
+            <Button
+              id="filter-button"
+              design="Transparent"
+              icon="filter"
+              tooltip={t('Graphs.colorizedTitle')}
```

Comment:
There’s a colon in the tooltip which we should remove.

Could also think of moving the text as a header of the popover. Another idea would be to call it something like "color by" because we're not really filtering or grouping (in a sense of rearranging content), are we? There are also other icons like sap-icon://legend or sap-icon://palette – just as an idea. Leaving it up to you ✌️